src/Parsers/Lexer.cpp

```diff
@@ -338,7 +338,7 @@ Token Lexer::nextTokenImpl()
             ++pos;
             if (pos < end && *pos == '|')
                 return Token(TokenType::Concatenation, token_begin, ++pos);
-            return Token(TokenType::ErrorSinglePipeMark, token_begin, pos);
+            return Token(TokenType::PipeMark, token_begin, pos);
```

Comment:
This will change behavior of `clickhouse` dialect. I'm not sure that we even can use the same lexer here - apparently the syntax of KQL is quite different, and this case is a good example...
@alexey-milovidov what do you think?

Reply:
Lexer can be extended and remain shared if the amount of changes is low and if the rest is compatible.
Although I'm not sure about the latter. I don't think that ClickHouse and Kusto share the same string escaping rules.

Reply:
Making the error message from Lexer slightly less precise is ok to me.

Reply:
@yakov-olkhovskiy , @alexey-milovidov ,  thank you very much, I will update soon.